lib/server.js

```diff
@@ -496,8 +499,9 @@ Server.prototype.use = function use() {
     var self = this;
     var handlers = Array.prototype.slice.call(arguments);
 
-    argumentsToChain(handlers).forEach(function forEach(h) {
-        self.chain.push(h);
+    argumentsToChain(handlers).forEach(function forEach(handler) {
+        handler._name = handler.name || 'use-' + self.useChain.count();
```

Comment:
Isn't this already happening in router?

Reply:
Router, pre, and use are registered separately with a different prefix.